octoprint_mrbeam/iobeam/onebutton_handler.py

```diff
@@ -401,36 +401,36 @@ def set_ready_to_laser(self, gcode_file=None):
         self._fireEvent(MrBeamEvents.READY_TO_LASER_START, dict(cpu_load=cpu_load))
         self._check_if_still_ready_to_laser()
 
-    def wait_for_cpu_below(self, cpu_max=70.0, check_interval=2.5):
+    def wait_for_cpu_below(self, cpu_max=60.0, check_interval=2.5):
         """
-        > ps aux | sort -nrk 3,3 | head -n 5
-        pi        3466 86.0 12.0 387884 105960 ?       Sl   15:49   1:22 /home/pi/oprint/bin/python /home/pi/oprint/bin/octoprint --host=127.0.0.1 --port=5000
-        root       925  5.3  1.5  29168 13524 ?        Sl   Apr12  83:33 /usr/bin/python /usr/local/bin/mrbeam_ledstrips
-        root       571  2.9  1.8  29328 16408 ?        Sl   Apr12  45:56 /usr/bin/python /usr/local/bin/netconnectd --config /etc/netconnectd.yaml --address /var/run/netconnectd.sock --logfile /var/log/netconnectd.log --foreground
-        root       436  1.2  2.8  97856 24668 ?        Ssl  Apr12  19:06 /usr/bin/python /usr/local/bin/iobeam
-        pi       23641  0.6  0.2   5108  2448 pts/1    S+   15:11   0:14 top
-        sort: write failed: standard output: Broken pipe
-        sort: write error
+        now it's actually not the cpu load, it's the cpu usage of the octoprint process.
+
+        this would be an alternative but it's downside is that it updates slower than top
+        # r"^\s*.+\s+\d+\s+([0-9.]+)\s+[0-9.]+\s+\d+\s+\d+\s+", re.MULTILINE
+        # command = "ps aux | sort -nrk 3,3 | head -n 5"
+        (seemingly it's a avg over that pas minute, documentation says it's the avg for the process since ever.)
         """
         cpu_load = -1.0
         find_cpu_load = re.compile(
-            r"^\s*.+\s+\d+\s+([0-9.]+)\s+[0-9.]+\s+\d+\s+\d+\s+", re.MULTILINE
+            r"^\s*\d+\s\w+\s+.*\s+[DRSTI]\s+([0-9.]+)\s+.*\s+octoprint$", re.MULTILINE
         )
+        command = "top -b -n 1 | grep octoprint"
+
         while cpu_load < 0.0 or cpu_load > cpu_max:
             try:
-                ps_out, code = exec_cmd_output(
-                    "ps aux | sort -nrk 3,3 | head -n 5", shell=True
-                )
-                self._logger.debug(
-                    "wait_for_cpu_below() test command output:\n%s", ps_out
-                )
-
+                ps_out, code = exec_cmd_output(command, shell=True)
+                # self._logger.debug(
+                #     "wait_for_cpu_below() test command output:\n%s", ps_out
+                # )
                 cpu_load = sum(map(lambda x: float(x), find_cpu_load.findall(ps_out)))
             except Exception:
                 self._logger.exception("Exception while reading cpu load: ")
                 time.sleep(check_interval)
             self._logger.info(
-                "wait_for_cpu_below() cpu_load: %s, code: %s", cpu_load, code
+                "wait_for_cpu_below() cpu_load: %s, cpu_max: %s, cpu_load > cpu_max: %s",
+                cpu_load,
+                cpu_max,
+                cpu_load > cpu_max,
```

Comment:
Please make it a debug log and without the function name

Reply:
ok